tests/lib/versa_test.py

```diff
@@ -0,0 +1,476 @@
+        'parameter-problem',
+        'redirect',
+        'redirect-message',
+        'router-advertisement',
+        'router-renumbering',
+        'router-solicit',
+        'router-solicitation',
+        'source-quench',
+        'time-exceeded',
+        'timestamp-reply',
+        'timestamp-request',
+        'unreachable',
+        'version-2-multicast-listener-report',
+    },
+}
+
+# Print a info message when a term is set to expire in that many weeks.
+# This is normally passed from command line.
+EXP_INFO = 2
+
+_IPSET = [nacaddr.IP('10.0.0.0/8'),
+          nacaddr.IP('2001:4860:8000::/33')]
+_IPSET2 = [nacaddr.IP('10.23.0.0/22'), nacaddr.IP('10.23.0.6/23', strict=False)]
+_IPSET3 = [nacaddr.IP('10.23.0.0/23')]
+_IPSET4 = [nacaddr.IP('10.0.0.0/20')]
+_IPSET5 = [nacaddr.IP('10.0.0.0/24')]
+
+
+class VersaTest(absltest.TestCase):
+
+  def setUp(self):
+    super().setUp()
+    self.naming = mock.create_autospec(naming.Naming)
+
+  def testHeaderComment(self):
+    pol = policy.ParsePolicy(GOOD_HEADER + ICMP_TYPE_TERM_0 , self.naming)
+    output = str(versa.Versa(pol, EXP_INFO))
+    self.assertIn('This is a test acl with a comment', output, output)
+
+  def testHeaderWithoutVersaHeader(self):
+    pol = policy.ParsePolicy(GOOD_HEADER_1 + ICMP_TYPE_TERM_0 , self.naming)
+    output = str(versa.Versa(pol, EXP_INFO))
+    self.assertIn('template _templatename', output, output)
+    self.assertIn('org-services _tenantname', output, output)
+    self.assertIn('access-policy-group _policyname', output, output)
+
+  def testHeaderWithoutVersaHeaderTemplate(self):
+    pol = policy.ParsePolicy(GOOD_HEADER_2 + ICMP_TYPE_TERM_0 , self.naming)
+    output = str(versa.Versa(pol, EXP_INFO))
+    self.assertIn('template test', output, output)
+
+  def testHeaderWithoutVersaHeaderTenant(self):
+    pol = policy.ParsePolicy(GOOD_HEADER_3 + ICMP_TYPE_TERM_0 , self.naming)
+    output = str(versa.Versa(pol, EXP_INFO))
+    self.assertIn('org-services tenant1', output, output)
+
+  def testHeaderWithNoVerbose(self):
+    pol = policy.ParsePolicy(GOOD_HEADER_NOVERBOSE+ ICMP_TYPE_TERM_0,
+                                            self.naming)
+    output = str(versa.Versa(pol, EXP_INFO))
+    self.assertNotIn('/*', output, output)
+    self.assertNotIn('*/', output, output)
+
+  def testHeaderWithoutVersaHeaderPolicy(self):
+    pol = policy.ParsePolicy(GOOD_HEADER_4 + ICMP_TYPE_TERM_0 , self.naming)
+    output = str(versa.Versa(pol, EXP_INFO))
+    self.assertIn('access-policy-group Default-Policy', output, output)
+
+  def testIcmpV4(self):
+    pol = policy.ParsePolicy(GOOD_HEADER_1 + ICMP_TYPE_TERM_0 , self.naming)
+    output = str(versa.Versa(pol, EXP_INFO))
+    self.assertIn(' predefined-services-list [ ICMP ]', output, output)
+
+  def testIcmpType(self):
+    pol = policy.ParsePolicy(GOOD_HEADER_1 + ICMP_TYPE_TERM_1 , self.naming)
+    self.assertRaises(versa.VersaUnsupportedTerm, versa.Versa, pol, EXP_INFO)
+
+  def testIcmpV6(self):
```

Comment:
We'll want to add some tests later to meet https://github.com/google/capirca/blob/master/doc/generator_patterns.md#icmp-and-icmpv6-handling

Reply:
will do so